### DISCUSSION TITLE
ZeppOS Dev

### DISCUSSION THREAD

**Discussion by chipset2011** [zepp-health/zeppos-forum#26] [Q&A] [answered]
Good afternoon, tell me what will happen to the web development environment, and why it was closed? Does anyone have detailed instructions on how to start development using VS Codium tools, I can't get a QR code, zeus-preview doesn't work...

**Comment by duanxiong** [comment]
It's recommended to use local development environment.

You can visit the offical website for more information about vscode plugin: https://developer.zepp.com/os/develop-vscode, and there are some videos on youtube: https://www.youtube.com/channel/UCU8LoAkoGotUNL82arnib5w, hope these can help you.

**Reply by chipset2011** [reply]
Attempting zeus-preview results in the message "The configuration of the device target needs to be preset in STATUS-build."
![image](https://user-images.githubusercontent.com/62571400/208767473-1bf74efa-0c32-4b94-aefb-f26a4bcaa672.png)

**Reply by chipset2011** [reply]
Trying to do a zeus-build leads to a build, but nothing happens when you click on the "open folder" button.
![image](https://user-images.githubusercontent.com/62571400/208768206-9f4576ce-f09b-4752-9288-8825a5b59d89.png)

**Reply by chipset2011** [reply]
I'm trying to put together your "Hello World" example

**Comment by mrcoon** [comment, marked as answer]
Before using the preview function, please select the target to be previewed.
![image](https://user-images.githubusercontent.com/10527392/208832390-654812b2-3c9b-4d37-afa5-19cd90822373.png)

**Comment by chipset2011** [comment]
It turned out to get a barcode, thank you, but now another problem, I tried to compile Hello, world for gtr3 PRO on gtr4 watch, but when trying to read the barcode it writes "download failed, error code: null value"

**Reply by chipset2011** [reply]
any ideas?
